go/vt/dbconfigs/dbconfigs.go

```diff
@@ -135,69 +135,102 @@ func registerPerUserFlags(dbc *userConfig, userKey string) {
 
 }
 
+// ConnParams contains mysql.connparams for now
+type ConnParams struct {
+	connParams *mysql.ConnParams
+}
+
+// MakeConnParams is for tests only
+func MakeConnParams() ConnParams {
+	return ConnParams{}
+}
+
+// GetConnParams returns the connections params
+func (cp ConnParams) GetConnParams() (*mysql.ConnParams, error) {
+	params, err := withCredentials(cp.connParams)
+	if err != nil {
+		return nil, err
+	}
+	return params, nil
+}
+
+// SetConnParams is to use for tests only
+func (cp *ConnParams) SetConnParams(mcp *mysql.ConnParams) {
```

Comment:
Since this is a read-only type, change all calls to `func (cp Connector)...`.
Here's some elaboration: https://flaviocopes.com/golang-methods-receivers/